docs/source/getting-started/what-is-jupyterhub.md

```diff
@@ -0,0 +1,465 @@
+- **SudoSpawner** is like LocalProcessSpawner but lets you run
+  JupyterHub without root.  `sudo` has to be configured to allow the
+  hub's user to run processes under other user IDs.
+
+- **SystemdSpawner** uses Systemd to start other processes.  It can
+  isolate users from each other and provide resource limiting.
+
+- **DockerSpawner** runs stuff in Docker, a containerization system.
+  This lets you fully isolate users, limit CPU, memory, and provide
+  other container images to fully customize the environment.
+
+- **KubeSpawner** runs on the Kubernetes, a cloud orchestration
+  system.  The spawner can easily limit users and provide cloud
+  scaling - but the spawner doesn't actually do that, Kubernetes
+  does.  The spawner just tells Kubernetes what to do.  If you want to
+  get KubeSpawner to do something, first you would figure out how to
+  do it in Kubernetes, then figure out how to tell KubeSpawner to tell
+  Kubernetes that.  Actually... this is true for most spawners.
+
+- **BatchSpawner** runs on computer clusters with batch job scheduling
+  systems (e.g Slurm, HTCondor, PBS, etc).  The user processes are run
+  as batch jobs, having access to all the data and software that the
+  users normally will.
+
+In short, spawners are the interface to the rest of the operating
+system, and to configure them right you need to know a bit about how
+the corresponding operating system service works.
+
+The spawner is responsible for the environment of the single-user
+notebook servers (described in the next section).  In the end, it just
+makes a choice about how to start these processes: for example, the
+Docker spawner starts a normal Docker container and runs the right
+command inside of it.  Thus, the spawner is responsible for setting
+what kind of software and data is available to the user.
+
+The spawner runs internally to the Hub process but communicates with
+outside services.  It is configured by `c.JupyterHub.spawner_class` in
+`jupyterhub_config.py`.
+
+If a user tries to launch a notebook server and it doesn't work, the
+error is usually with the spawner or the notebook server (as described
+in the next section).  Each spawner outputs some logs to the main
+JupyterHub logs, but may also have logs in other places depending on
+what services it interacts with (for example, the Docker spawner
+somehow puts logs in the Docker system services, Kubernetes through
+the `kubectl` API).
+
+
+### Proxy
+
+Previously, we said that the hub is between the user's web browser and
+the user's notebook servers.  It actually isn't directly between,
+because the JupyterHub **proxy** relays connections between the users
+and their single-user notebook servers.  What this basically means is
+that the hub itself can shut down and the proxy can continue to
+allow users to communicate with their notebook servers.  (This
+further emphasizes that the hub is responsible for starting, not
+running, the notebooks).  By default, the hub starts the proxy
```

Comment:
I think it would be helpful if we began this document with a diagram that had one "box" for each major component we describe in this guide